src/stdlib_experimental_quadrature_simps.fypp

```diff
@@ -0,0 +1,277 @@
+                integral = dx/3.0_${k1}$*(y(1) + 4*sum(y(2:n-1:2)) + 2*sum(y(3:n-2:2)) + y(n))
+            else
+                if (present(even)) then
+                    if (even < 0) then
+                        ! 3/8th rule on left
+                        integral = simps38(y(1:4), dx) + simps(y(4:n), dx)
+                        return
+                    else if (even > 0) then
+                        ! 3/8 rule on right
+                        integral = simps(y(1:n-3), dx) + simps38(y(n-3:n), dx)
+                        return
+                    else
+                        ! fall through
+                    end if
+                end if
+                ! either `even` not present or is zero
+                ! equivalent to averaging left and right
+                integral = dx/48.0_${k1}$ * (17*(y(1) + y(n)) + 59*(y(2) + y(n-1)) &
+                        + 43*(y(3) + y(n-2)) + 49*(y(4) + y(n-3)) + 48*sum(y(5:n-4)))
+            end if
+        end select
+    end function simps_dx_${k1}$
+
+#:endfor
+#:for k1, t1 in REAL_KINDS_TYPES
+
+    pure recursive module function simps_x_${k1}$(y, x, even) result(integral)
+        ${t1}$, dimension(:), intent(in) :: y
+        ${t1}$, dimension(size(y)), intent(in) :: x
+        integer, intent(in), optional :: even
+        ${t1}$ :: integral
+
+        integer :: i
+        integer :: n
+
+        ${t1}$ :: h1, h2, h3
+        ${t1}$ :: a, b, c, d
+
+        n = size(y)
+
+        select case (n)
+        case (0:1)
+            integral = 0.0_${k1}$
+        case (2)
+            integral = 0.5_${k1}$*(x(2) - x(1))*(y(1) + y(2))
+        case (3)
+            h1 = x(2) - x(1)
+            h2 = x(3) - x(2)
+            a = (2*h1**2 + h1*h2 - h2**2)/(6*h1)
+            b = (h1+h2)**3/(6*h1*h2)
+            c = (2*h2**2 + h1*h2 - h1**2)/(6*h2)
+            integral = a*y(1) + b*y(2) + c*y(3)
+        case (4)
+            integral = simps38(y, x)
+        ! case (6) unneeded?
+        case default
+            if (mod(n, 2) == 1) then
+                integral = 0.0_${k1}$
+                do i = 1, n-2, 2
+                    h1 = x(i+1) - x(i)
+                    h2 = x(i+2) - x(i+1)
+                    a = (2*h1**2 + h1*h2 - h2**2)/(6*h1)
+                    b = (h1+h2)**3/(6*h1*h2)
+                    c = (2*h2**2 + h1*h2 - h1**2)/(6*h2)
+                    integral = integral + a*y(i) + b*y(i+1) + c*y(i+2)
+                end do
+            else
+                if (present(even)) then
+                    if (even < 0) then
+                        ! 3/8 rule on left
+                        integral = simps38(y(1:4), x(1:4)) + simps(y(4:n), x(4:n))
+                        return
+                    else if (even > 0) then
+                        ! 3/8 rule on right
+                        integral = simps(y(1:n-3), x(1:n-3)) + simps38(y(n-3:n), x(n-3:n))
+                        return
+                    else
+                        ! fall through
```

Comment:
```suggestion

```

Reply:
What is the suggestion here?

Reply:
These two lines are not needed and can be removed.

Reply:
I think it helps make the control flow easier to understand. Compilers will eliminate it anyway.